core_impl/src/test/java/com/google/instrumentation/trace/SpanImplTest.java

```diff
@@ -63,15 +70,47 @@ public void toSpanData_NoRecordEvents() {
             SPAN_NAME,
             parentSpanId,
             false,
+            TraceParams.DEFAULT,
             startEndHandler,
             timestampConverter,
             testClock);
+    span.addAnnotation(Annotation.fromDescription(ANNOTATION_DESCRIPTION));
+    span.addAnnotation(ANNOTATION_DESCRIPTION, attributes);
+    span.addNetworkEvent(NetworkEvent.builder(NetworkEvent.Type.RECV, 1).setMessageSize(3).build());
+    span.addLink(Link.fromSpanContext(spanContext, Link.Type.CHILD));
     span.end();
     exception.expect(IllegalStateException.class);
     span.toSpanData();
   }
 
   @Test
+  public void noEventsRecordedAfterEnd() {
+    SpanImpl span =
+        SpanImpl.startSpan(
+            spanContext,
+            recordSpanOptions,
+            SPAN_NAME,
+            parentSpanId,
+            false,
+            TraceParams.DEFAULT,
+            startEndHandler,
+            timestampConverter,
+            testClock);
+    span.end();
+    span.addAnnotation(Annotation.fromDescription(ANNOTATION_DESCRIPTION));
+    span.addAnnotation(ANNOTATION_DESCRIPTION, attributes);
+    span.addNetworkEvent(NetworkEvent.builder(NetworkEvent.Type.RECV, 1).setMessageSize(3).build());
+    span.addLink(Link.fromSpanContext(spanContext, Link.Type.CHILD));
+    SpanData spanData = span.toSpanData();
+    assertThat(spanData.getStartTimestamp()).isEqualTo(timestamp);
+    assertThat(spanData.getAnnotations().getEvents().size()).isEqualTo(0);
```

Comment:
Suggestion: You could write `assertThat(spanData.getAnnotations().getEvents()).isEmpty();`.